core/config/project_settings.cpp

```diff
@@ -435,23 +436,45 @@ void ProjectSettings::_get_property_list(List<PropertyInfo> *p_list) const {
 		if (v->restart_if_changed) {
 			vc.flags |= PROPERTY_USAGE_RESTART_IF_CHANGED;
 		}
-		vclist.insert(vc);
-	}
 
-	for (const _VCSort &E : vclist) {
-		String prop_info_name = E.name;
-		int dot = prop_info_name.find_char('.');
-		if (dot != -1 && !custom_prop_info.has(prop_info_name)) {
-			prop_info_name = prop_info_name.substr(0, dot);
+		int dot = vc.name.rfind_char('.');
+		if (dot != -1 && !custom_prop_info.has(vc.name)) {
+			StringName n = vc.name.substr(0, dot);
+			if (props.has(n)) { // Property is an override.
+				if (!setting_overrides.has(n)) {
+					Vector<_VCSort> vvc;
+					setting_overrides.insert(n, vvc);
+				}
+				setting_overrides.get(n).append(vc);
```

Comment:
You can do that:
```suggestion
				setting_overrides[n].append(vc);
```
HashMap supports it.